lib/matplotlib/backends/backend_ps.py

```diff
@@ -1499,30 +1504,28 @@ def gs_distill(tmpfile, eps=False, ptype='letter', bbox=None, rotated=False):
     outfile = tmpfile + '.output'
     dpi = rcParams['ps.distiller.res']
 
-
     gs_exe = ps_backend_helper.gs_exe
     if ps_backend_helper.supports_ps2write: # gs version >= 9
         device_name = "ps2write"
     else:
         device_name = "pswrite"
 
-    command = '%s -dBATCH -dNOPAUSE -r%d -sDEVICE=%s %s -sOutputFile="%s" \
-                "%s" > "%s"'% (gs_exe, dpi, device_name,
-                               paper_option, psfile, tmpfile, outfile)
-
+    command = [str(gs_exe), "-dBATCH", "-dNOPAUSE", "-r%d" % dpi,
+               "-sDEVICE=%s" % device_name, paper_option,
+               "-sOutputFile=%s" % psfile, tmpfile]
     verbose.report(command, 'debug')
-    exit_status = os.system(command)
-
-    with io.open(outfile, 'rb') as fh:
-        if exit_status:
-            output = fh.read()
-            m = "\n".join(["ghostscript was not able to process your image.",
-                           "Here is the full report generated by ghostscript:",
-                           "",
-                           "%s"])
-            # use % to prevent problems with bytes
-            raise RuntimeError(m % output)
-        else:
+    try:
+        with open(outfile, "w") as fout:
+            subprocess.check_call(command,
```

Comment:
That's because in the case of an exception, nothing is actually assigned to `output`.